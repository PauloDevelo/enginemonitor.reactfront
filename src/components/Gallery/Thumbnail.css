.thumbnail {
    margin: 1px;
    border-radius: inherit;
    transition:all 0.3s ease;
}

.grow:hover
{
        -webkit-transform: scale(1.05);
        -ms-transform: scale(1.05);
        transform: scale(1.05);
}