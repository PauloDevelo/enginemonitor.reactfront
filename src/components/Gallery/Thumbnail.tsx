// eslint-disable-next-line no-use-before-define
import React from 'react';
import Img from '../ImageComponent/Img';
import storageService from '../../services/StorageService';

import Loading from '../Loading/Loading';

// eslint-disable-next-line no-unused-vars
import { ImageModel } from '../../types/Types';
import './Thumbnail.css';

type Props = {
    image: ImageModel;
    onClickImage: undefined | (() => void);
}
function Thumbnail({ image, onClickImage }: Props) {
  const displayImage = () => {
    if (onClickImage !== undefined) {
      onClickImage();
    }
  };

  const alt = `${image.title} - ${image.description}`;

  return <Img storage={storageService.getUserStorage()} src={image.thumbnailUrl} alt={alt} onClick={displayImage} className="thumbnail grow" loader={<span className="spinnerContainer"><Loading onClick={displayImage} /></span>} />;
}

export default React.memo(Thumbnail);
