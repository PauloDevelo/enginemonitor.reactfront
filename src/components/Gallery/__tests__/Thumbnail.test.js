import React from 'react';
import { mount } from 'enzyme';

import Thumbnail from '../Thumbnail';
import updateWrapper from '../../../testHelpers/EnzymeHelper';

describe('Component Thumbnail', () => {
  it('should render an image using the thumbnail url', () => {
    // Arrange
    const image = {
      url: 'http://localhost:3000/image.jpeg',
      thumbnailUrl: 'http://localhost:3000/thumbnail.jpeg',
      parentUiId: 'parentUiId',
      title: 'a title',
      description: 'a desc',
      sizeInByte: '1234',
    };

    const onClick = jest.fn();

    // Act
    const wrapper = mount(<Thumbnail image={image} onClickImage={onClick} />);

    // Assert
    expect(wrapper.find('Img').hasClass('thumbnail')).toEqual(true);
    expect(wrapper.find('Img').hasClass('grow')).toEqual(true);
    expect(wrapper.find('Img').props().src).toEqual(image.thumbnailUrl);
    expect(wrapper.find('Img').props().alt).toEqual(`${image.title} - ${image.description}`);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render a clickable image', async () => {
    // Arrange
    const image = {
      url: './images.engine.svg',
      thumbnailUrl: './images.engine.svg',
      parentUiId: 'parentUiId',
      title: 'a title',
      description: 'a desc',
      sizeInByte: '1234',
    };

    const onClick = jest.fn();
    const wrapper = mount(<Thumbnail image={image} onClickImage={onClick} />);
    await updateWrapper(wrapper);

    // Act
    wrapper.find('Loading').simulate('click');

    // Assert
    expect(onClick.mock.calls.length).toEqual(1);
  });
});
