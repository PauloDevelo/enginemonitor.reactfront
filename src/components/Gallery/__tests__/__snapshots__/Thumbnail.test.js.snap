// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Thumbnail should render an image using the thumbnail url 1`] = `
<Memo(Thumbnail)
  image={
    Object {
      "description": "a desc",
      "parentUiId": "parentUiId",
      "sizeInByte": "1234",
      "thumbnailUrl": "http://localhost:3000/thumbnail.jpeg",
      "title": "a title",
      "url": "http://localhost:3000/image.jpeg",
    }
  }
  onClickImage={[MockFunction]}
>
  <Img
    alt="a title - a desc"
    className="thumbnail grow"
    container={[Function]}
    decode={true}
    loader={
      <span
        className="spinnerContainer"
      >
        <Loading
          onClick={[Function]}
        />
      </span>
    }
    onClick={[Function]}
    src="http://localhost:3000/thumbnail.jpeg"
    unloader={false}
  >
    <span
      className="spinnerContainer"
    >
      <Loading
        onClick={[Function]}
      >
        <div
          className="lds-spinner"
          onClick={[Function]}
        >
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
        </div>
      </Loading>
    </span>
  </Img>
</Memo(Thumbnail)>
`;
